src/hooks/useScrollDirection.ts

```diff
@@ -0,0 +1,46 @@
+import { useCallback, useEffect, useState } from 'react';
+
+export enum ScrollDirection {
+  Down = 'down',
+  Up = 'up',
+}
+
+/**
+ * A hook that detects scrolling and determines whether
+ * the user is scrolling up and down and passes it to the
+ * callback.
+ *
+ * @param {direction: ScrollDirection, position: number) => void} onDirectionChange
+ */
+const useScrollDirection = (
+  onDirectionChange: (direction: ScrollDirection, position: number) => void,
+) => {
+  const [, setLastYPosition] = useState<number>(0);
```

Comment:
also,
disclaimer: I don't really have crystal clear picture about the thing I will say. 😁

Does it make sense to use `useRef` here? 

```
const lastYPosition = useRef(0);

// accessing the position 
lastYPosition.current

// set lastYPosition
lastYPosition.current = newYPosition
```

I remember I need to use `useRef` a couple times in the past to deal with heave animation

Essentially, both `useRef` and `useState` can store a value. But the difference is with `useState`, everytime the state is updated. It will trigger `re rendering` with `useRef` it doesn't trigger re rendering. Which is what we want in this context. 

We don't want changing y position trigger re render, we only want to call `onDirectionChange` callback when new direction is detected.

CMIIW. Because I think I don't have as much as context as you for this code 😄 

Reference 
- https://blog.logrocket.com/usestate-vs-useref/

Reply:
Thanks a lot @muhajirdev for the suggestion. It seems using `useRef` in this case works better than `useState`. Will update the code accordingly.